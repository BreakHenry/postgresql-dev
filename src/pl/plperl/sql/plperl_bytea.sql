
set plperl.pass_binary_bytea = true;

create function test_perl_bytea(str bytea) 
returns bytea
language plperl as
$func$

my $arg = shift;
my $enc = $arg;
$enc =~ s/(.)/sprintf("%.2x",ord($1))/egs;
elog(NOTICE,"encoded: $enc");
return $arg;

$func$;

select test_perl_bytea('\x01020304f1f2f3f4');
