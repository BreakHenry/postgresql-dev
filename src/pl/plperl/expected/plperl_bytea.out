create function test_perl_bytea(str bytea) 
returns bytea
language plperl as
$func$

my $arg = shift;
my $enc = $arg;
$enc =~ s/(.)/sprintf("%.2x",ord($1))/egs;
elog(NOTICE,"encoded: $enc");
return $arg;

$func$;
select test_perl_bytea('\x01020304f1f2f3f4');
NOTICE:  encoded: 01020304f1f2f3f4
CONTEXT:  PL/Perl function "test_perl_bytea"
  test_perl_bytea   
--------------------
 \x01020304f1f2f3f4
(1 row)

